### PR TITLE
Migrate some leftovers to "network-timeout" settings key

### PR DESCRIPTION
**Pull request by pathmapper**
`qgis/networkAndProxy/networkTimeout` is deprecated, ref https://github.com/qgis/QGIS/issues/53759#issuecomment-1627623824

### CODE REVIEW

**Comment by pathmapper** [comment]
ping @3nids

File: tests/src/providers/testqgswcspublicservers.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ TestQgsWcsPublicServers::TestQgsWcsPublicServers( const QString &cacheDirPath, i
 TestQgsWcsPublicServers::~TestQgsWcsPublicServers()
 {
   QgsSettings settings;
-  settings.setValue( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), mOrigTimeout );
+  settings.setValue( QStringLiteral( "network/network-timeout" ), mOrigTimeout );
```
Comment (3nids) [inline tests/src/providers/testqgswcspublicservers.cpp:61]:
```suggestion
  QgsNetworkAccessManager::settingsNetworkTimeout->setValue( mOrigTimeout );
```

File: tests/src/providers/testqgswcspublicservers.cpp [hunk 2]
```diff
@@ -69,8 +69,8 @@ void TestQgsWcsPublicServers::init()
   // Unfortunately this seems to be the only way to set timeout, we try to reset it
   // at the end but it can be canceled before ...
   QgsSettings settings;
-  mOrigTimeout = settings.value( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), "60000" ).toInt();
-  settings.setValue( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), mTimeout );
+  mOrigTimeout = settings.value( QStringLiteral( "network/network-timeout" ), "60000" ).toInt();
```
Comment (3nids) [inline tests/src/providers/testqgswcspublicservers.cpp:72]:
```suggestion
  mOrigTimeout = QgsNetworkAccessManager::settingsNetworkTimeout->value().toInt();
```

File: src/auth/oauth2/core/qgsauthoauth2method.cpp [hunk 3]
```diff
@@ -190,7 +190,7 @@ bool QgsAuthOAuth2Method::updateNetworkRequest( QNetworkRequest &request, const
 
 
     QgsSettings settings;
-    const QString timeoutkey = QStringLiteral( "qgis/networkAndProxy/networkTimeout" );
+    const QString timeoutkey = QStringLiteral( "network/network-timeout" );
     const int prevtimeout = settings.value( timeoutkey, QStringLiteral( "-1" ) ).toInt();
     const int reqtimeout = o2->oauth2config()->requestTimeout() * 1000;
     settings.setValue( timeoutkey, reqtimeout );
```
[the remark this answers is not in the document]
Comment (pathmapper) [reply src/auth/oauth2/core/qgsauthoauth2method.cpp]: 👍  -> https://github.com/qgis/QGIS/pull/53763/commits/9fb86421215c748e90a1af44631ecb3358af024b

**Comment by pathmapper** [comment]
Thanks for the hints @3nids, updated the PR accordingly.

**Review by elpaso — APPROVED** [review]

**Pull request merged**